src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-only",
+        action="store_true",
+        dest="print_deps_only",
+        default=False,
+        help="Deprecated, equivalent to `--print-deps-to -`",
+    )
+    parser.add_argument(
+        "--print-deps-to",
+        "--print-deps-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the dependencies to the given file "
+        + "(use `-` for stdout)",
+    )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the  names of the required extras to the given file "
+        + "(use `-` for stdout)",
+    )
+
+
+@impl
+def tox_register_tox_env(register):
+    register.add_run_env(CurrentEnv)
+    register.add_run_env(PrintEnv)
+
+
+@impl
+def tox_add_core_config(core_conf, config):
+    if config.options.current_env:
+        config.options.default_runner = "current-env"
+        return
+
+    if config.options.print_deps_only:
+        warnings.warn(
+            "--print-deps-only is deprecated; use `--print-deps-to -`",
+            DeprecationWarning,
+        )
+        if not config.options.print_deps_to:
+            config.options.print_deps_to = sys.stdout
+        else:
+            raise RuntimeError(
+                "--print-deps-only cannot be used together " "with --print-deps-to"
+            )
+
+    if getattr(config.options.print_deps_to, "name", object()) == getattr(
+        config.options.print_extras_to, "name", object()
+    ):
+        raise RuntimeError(
+            "The paths given to --print-deps-to and --print-extras-to cannot be identical."
+        )
+
+    if config.options.print_deps_to or config.options.print_extras_to:
+        config.options.default_runner = "print-env"
+        return
+
+    # No options used - switch back to the standard runner
+    # Workaround for: https://github.com/tox-dev/tox/issues/2264
+    config.options.default_runner = "virtualenv"
+
+
+class CurrentEnv(PythonRun):
+    def __init__(self, create_args):
+        self._executor = None
+        self._installer = None
+        self._path = []
+        super().__init__(create_args)
+
+    @staticmethod
+    def id():
+        return "current-env"
+
+    @property
+    def _default_package_tox_env_type(self):
+        return None
+
+    @property
+    def _external_pkg_tox_env_type(self):
+        return None
+
+    @property
+    def _package_tox_env_type(self):
+        return None
+
+    @property
+    def executor(self):
+        if self._executor is None:
+            self._executor = CurrentEnvRunExecutor(self.options.is_colored)
```

Comment:
A nicer solution would be to just set allowlist_external to be '*' for all current env, you can do this by injecting a memory loader within https://tox.wiki/en/rewrite/plugins.html#tox.plugin.spec.tox_add_env_config see https://github.com/tox-dev/tox/blob/rewrite/src/tox/session/cmd/devenv.py#L33-L41

Reply:
Nice, one more thing I can remove!

Reply:
It seems that it's not possible to call `get_env` inside `tox_add_env_config` because it causes an endless recursion. How can I overwrite the value without creating a whole new config? Item assignment is not possible.

Reply:
Is there any other way how to do it please?

Reply:
@frenzymadness can you open an issue under tox-dev/tox for this?

Reply:
Sure thing, here it is: https://github.com/tox-dev/tox/issues/2287